### PR TITLE
Check in generated struct info rather than auto-generating it in the sysroot. NFC

### PR DESCRIPTION
**Pull request by sbc100**
I noticed while working `gen_sig_info.py` in #18985 that is was simpler to just keep a copy of the generated file checked into source control. We can use a unit test check that its up-to-date on each commit.  In fact we already had a unit test that was doing this and keeping a copy of the expected output in the test directory.

This simplifies the emcc linker code since we can always assume the presence of the struct info file.

### CODE REVIEW

**Comment by kripken** [comment]
IIUC this means that we need to manually run that `gen_struct_info` script when changes are needed in that file, is that right? (Compared to before where it was automatic, at least if the cache was cleared.) If so that might be worth documenting in `docs/` and/or https://emscripten.org/docs/contributing/developers_guide.html ?

**Comment by sbc100** [comment]
> IIUC this means that we need to manually run that `gen_struct_info` script when changes are needed in that file, is that right? (Compared to before where it was automatic, at least if the cache was cleared.) If so that might be worth documenting in `docs/` and/or https://emscripten.org/docs/contributing/developers_guide.html ?

Right, although for local development you still had to remember to do something like `./emcc --clear-cache`.. since we didn't have any kind of dependency system that would know when re-generation was needed.

I'll update the docs.

File: site/source/docs/contributing/developers_guide.rst [hunk 1]
```diff
@@ -151,10 +151,19 @@ one of them, with the others kept fixed). Doing this will require rebuilding
 locally, which was not needed in the main bisection described in this
 section.
 
+Working with C structs and defines
+==================================
+
+If you change the layout of C structs or modify C defines that are used in
+JavaScript library files you may need to modify ``tools/struct_info.json``.  Any
+time that file is modified of a struct layout is changed you will need to run
```
Comment (kripken) [inline site/source/docs/contributing/developers_guide.rst:159]:
```suggestion
time that file is modified or a struct layout is changed you will need to run
```

File: test/test_other.py [hunk 2]
```diff
@@ -11803,12 +11803,14 @@ def test_split_main_module(self):
     self.assertIn('Hello from lib!', result)
 
   def test_gen_struct_info(self):
-    # This tests is fragile and will need updating any time any of the referenced
-    # structs or defines change.   However its easy to rebaseline with
-    # --rebaseline and it prevents regressions or unintended changes
-    # to the output json.
+    # This test will start failing whenever the struct info changes (e.g. offset or defines
+    # change).  However its easy to rebaseline with --rebaseline.
```
Comment (kripken) [inline test/test_other.py:11807]:
```suggestion
    # change).  However it's easy to rebaseline with --rebaseline.
```

File: test/test_other.py [hunk 3]
```diff
@@ -11803,12 +11803,14 @@ def test_split_main_module(self):
...
     self.run_process([PYTHON, path_from_root('tools/gen_struct_info.py'), '-o', 'out.json'])
```
Comment (kripken) [inline test/test_other.py:11808]: Wait, how does rebaseline work? Doesn't the code here need to use this pattern?
```python
if common.EMTEST_REBASELINE:
  # generate the file
```

File: tools/gen_struct_info.py [hunk 4]
```diff
@@ -278,8 +278,6 @@ def inspect_headers(headers, cflags):
   # Run the compiled program.
   show('Calling generated program... ' + js_file[1])
   args = []
-  if settings.MEMORY64:
-    args += shared.node_bigint_flags()
```
Comment (kripken) [inline tools/gen_struct_info.py:282]: why is this no longer needed?

File: test/test_other.py (hunk 3 above)
Comment (sbc100) [reply test/test_other.py:11808]: `assertFileContents` automatically takes care of that.

**Review by kripken — APPROVED** [review]

File: test/test_other.py (hunk 3 above)
Comment (kripken) [reply test/test_other.py:11808]: Oh, cool!

**Pull request merged**